src/python/pants/backend/awslambda/python/lambdex.py

```diff
@@ -9,7 +9,4 @@ class Lambdex(PythonToolBase):
 
     options_scope = "lambdex"
     default_version = "lambdex==0.1.3"
-    # TODO(John Sirois): Remove when we can upgrade to a version of lambdex with
-    # https://github.com/wickman/lambdex/issues/6 fixed.
-    default_extra_requirements = ["setuptools==49.2.0"]
```

Comment:
That TODO isn't resolved, I think we still need setuptools.

Instead, we either want to
1) Keep the upgrade to `>=49.2.1,<49.3` and add the class property `default_interpreter_constraints = ["CPython>=3.6"]`
2) Change the version to `<45` so that it works with Python 2.